src/main/java/io/vertx/core/http/WebSocketBase.java

```diff
@@ -136,6 +137,31 @@
   WebSocketBase writeTextMessage(String text);
 
   /**
+   * Writes a ping to the connection. This will be written in a single frame, so the contents of data should be small.
+   * This method should not be used to write application data and should only be used for implementing a keep alive or
+   * to ensure the client is still responsive, see RFC 6455 Section 5.5.2.
+   *
+   * @param data the data to write
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Fluent
+  WebSocketBase writePing(Buffer data);
+
+  /**
+   * Writes a pong to the connection. This will be written in a single frame, so the contents of data should be small.
+   * This method should not be used to write application data and should only be used for implementing a keep alive or
+   * to ensure the client is still responsive, see RFC 6455 Section 5.5.2. RFC 6455 Secion 5.5.3 states that pongs may
+   * be sent unsolicited in order to implement a one way heartbeat.
+   *
+   * @param data the data to write
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Fluent
+  WebSocketBase writePong(Buffer data);
```

Comment:
I think we should be able to write a ping but not a pong.

Pong should be handled by the internal implementation when it receives a ping.

Reply:
I exposed this method because this behavior is specifically allowed in RFC 6455 5.5.3 paragraph 4:

```
A Pong frame MAY be sent unsolicited.  This serves as a unidirectional heartbeat.  A response to an unsolicited Pong frame is not expected.
```

I can't imagine writing an application that only sends Pong so I'm in favor of removing it.  I'll remove it there is agreement that we should despite the specification.

Reply:
Add a note saying that the websocket internally takes care of sending pong for ping and that the user should not use it for this purpose.

Reply:
I will also write a note about the 125Byte size limit :)